src/main/kotlin/Car.kt

```diff
@@ -0,0 +1,10 @@
+class Car(val wantsToDrive: Boolean, val roadNetwork : Network, var delayed : Boolean? = null) {
```

Comment:
Dealing with `null` is always a little hairy. I think it stands to reason that a car is initially not delayed, so setting to `false` should be ok - your choice, though.

Also, I think `delayed` does not need to be in the constructor. Why not declare it like [this](https://github.com/nicolaiparlog/energy-model/blob/master/src/main/kotlin/org/codefx/demo/bingen/bank/Bank.kt#L7-L8)?